### ISSUE TITLE
iRT Bus Connection not working

### ISSUE THREAD

**Issue by tbergmair** [Victor-Mo/IRT-EMS-ESP#10]
**Question**
I ordered a iRT Board that came today - flashed it with the latest bin 1.9.10 , got wifi running, connected it to my iRT Bus , set passive mode in Config, rebooted , telnet the device, set again to mode 4 , restart, set log j - then see the output

**Screenshots**
[TELNET] Connected to IRT-ESP version 1.9.10.20110420. Type ? for commands.
log j

System Logging set to Jabber mode

->
(00:53:50.834) irt_rawTelegram: 03: F8 78 80
(00:53:50.835) irt_crcErr1: 03: F8 78 80
Error! Unable to read the iRT bus.
(00:54:04.236) irt_rawTelegram: 03: F8 78 80
(00:54:04.236) irt_crcErr1: 03: F8 78 80

Any suggestions what i can do ?

br thomas

**Comment by Victor-Mo** [comment]
Hi Thomas,

Can you provide me with a little bit more information ? What boiler do you have ? What thermostat and how did you connect the interface board ?

Thanks,

Victor

**Comment by tbergmair** [comment]
Hi Viktor!

Thanks for reacting!  

I did more research and test the last days with some progress

I got a Buderus GB112  with a MEC and a HW4201,  i connected it now to ECO-BUS - Pin 2+3 ( as 1 seems to be ground),
the Webinterface still says "EMS Bus is not connected."  , if I telnet the device and write command "log j"  i get now the following output : 

* IRT-ESP version 1.9.10.20110420
[FS] System config loaded (501 bytes)
[FS] Custom config loaded (318 bytes)
[WEB] Web server started
[UART] Opened Rx/Tx connection (iRT)
[WIFI] Connecting to SSID: tb_ap...
[WIFI] Connected to SSID tb_ap (hostname: irt-esp, IP: 10.0.1.233)
[OTA] Listening to firmware updates on irt-esp.local:8266
[SYSTEM] Disabling serial port communication
Error! Unable to read the iRT bus.
Error! Unable to read the iRT bus.
Error! Unable to read the iRT bus.
Error! Unable to read the iRT bus.
[TELNET] Connected to IRT-ESP version 1.9.10.20110420. Type ? for commands.
log j
(00:03:26.336) irt_otherTelegram: 40: A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66
(00:03:26.540) irt_rawTelegram: 40: A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66
(00:03:26.543) irt_otherTelegram: 40: A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66
(00:03:26.748) irt_rawTelegram: 40: A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66
(00:03:26.751) irt_otherTelegram: 40: A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66

br
thomas

**Comment by jperquin** [comment]
Your heater looks pretty similar to the Nefit that we have. I took this picture from the manual I found online at: http://myskill.be/Buderus%20GB1%20technisch.PDF

![image](https://user-images.githubusercontent.com/46850501/103922313-dda8fc80-5113-11eb-97c5-44678e17c81a.png)

The wires from the EMS-iRT board go to ports 3 and 4. For my set-up to work correctly I had to disconnect the bridge-wire on ports 1 and 2..

If I am correct, the MEC and HW4201 is your (wireless?) thermostat. You will not need these to operate the EMS-iRT board. Victor knows more about this than me, but in most cases the EMS-iRT board does not like to be sharing resources with old thermostats..

**Comment by tbergmair** [comment]
Hi !  the heater maybe ( i don't know the nefit)  - but i connected it to the eco-bus on the HW4201 - the control unit of all!  I tried also to remove the Roomthermostat MEC while testing - but it made no difference to the log output

**Comment by tbergmair** [comment]
if i would only connect the irt board on the heater it would be no use at all when i would therefor need to disconnect the HW4201 - as all the thermostats and pumps are connect to the HW4201

**Comment by jperquin** [comment]
Maybe share a schematic with the connections (incl port numbers) of all components?
That may provide insight as to how these components (GB112, MEC, HW4201) interact..

**Comment by Victor-Mo** [comment]
@tbergmair I have done some research on your setup. The interface only works on the iRT bus between the boiler (GB112) and the control unit (HW4201). From what I could find the ECO-BUS is some sort of CAN bus ( https://github.com/flyingflo/logamatic ).

I did found the following connection diagram:
![HW4201_UBA](https://user-images.githubusercontent.com/56435760/103940870-5b7a0180-512e-11eb-9f87-b271c5733e0b.png)

I think the best place to connect the interface is on the HW4201 on pins 1 and 2 of the UBA connection. But anywhere between the boiler and control unit is fine. But I do find it a very 'strange' how the boiler and the control unit are connected. Is this also how your setup is ? Or has it a different connection.

Your setup is very similar to the setup in issue #7 It might be a good source of information.

**Comment by tbergmair** [comment]
Hi Victor !
Yes - thats the way all is connected !  Did u see my posting from yesterday ?  I connected it to Port 2 and 3 of the ECO-BUS - and at least sometimes I got some messages 

I can try to connect it in parallel to the UBA Ports - but i CAN NOT disconnect the HW4201 from the Boiler as then NOTING would be working as all the Sensors and Pumps are connected to HW4201 !

**Comment by Victor-Mo** [comment]
Hi Thomas,

From what i could find on the internet I pieced together a picture of what I think is your setup:
![hw4201_mec_passive_4](https://user-images.githubusercontent.com/56435760/104033506-e3fcae80-51cf-11eb-95eb-3985576abb27.png)

I also include where the interface board should be connected.

**Comment by tbergmair** [comment]
many thanks - I will try that tomorrow !

**Comment by tbergmair** [comment]
I tried to connect it that way - but it didn't get any better - i only got "Error! Unable to read the iRT bus."  - not even the messages i got on eco-bus in my post from 4days ago

**Comment by Victor-Mo** [comment]
Did the 'log j' show any output ? Did you try swapping the two wires to the interface board ?

**Comment by tbergmair** [comment]
yes- tried both !

i only get 👍 
->
log j

System Logging set to Jabber mode

->
Error! Unable to read the iRT bus.
Error! Unable to read the iRT bus.


when i connected it to eco-bus 2+3 i at least got :


log j
(00:03:26.336) irt_otherTelegram: 40: A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66
(00:03:26.540) irt_rawTelegram: 40: A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66
(00:03:26.543) irt_otherTelegram: 40: A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66
(00:03:26.748) irt_rawTelegram: 40: A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66
(00:03:26.751) irt_otherTelegram: 40: A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66 A0 53 DD 66

**Comment by Victor-Mo** [comment]
I think the problem is the 'strange' way the controller is connected. Normally the thermostat is connected at points 3 and 4. When I can find sometime i will build a setup with the thermostat connected to 5 and 10 and see if I can detect a correct signal.

**Comment by iamceph** [comment]
Hey guys, I got the exact same problem. My boiler is Nefit EcomLine HR43. I connected the device to the 3 and 4, same as the thermostat is and I got the exat same output. Here is my log message: https://paste.gg/p/iamceph/79c45e6adbd54dc89bf17e57d8a582fa

I did not wanted to create another issue, if I should, just tell me.
Thanks for your time! :)

**Comment by jperquin** [comment]
The Nefit Ekomline should work "right out the box".

A few things you might want to try:

- disconnect wire between ports 1 and 2.
- disconnect regular thermostat in case you are running that in parellel.
- change to settings 5(active with polling).

**Comment by iamceph** [comment]
Thank you, will try

**Comment by iamceph** [comment]
> The Nefit Ekomline should work "right out the box".
> 
> A few things you might want to try:
> 
> * disconnect wire between ports 1 and 2.
> * disconnect regular thermostat in case you are running that in parellel.
> * change to settings 5(active with polling).

So I tried these steps and now it seems that communication is offline.

Log:
Trying 192.168.1.104...
Connected to 192.168.1.104.
Escape character is '^]'.


* IRT-ESP version 1.9.10.20110420
[FS] System config loaded (509 bytes)
[FS] Custom config loaded (317 bytes)
[WEB] Web server started
[UART] Opened Rx/Tx connection (iRT)
[WIFI] Connecting to SSID: PCR_sledovani_24...
[WIFI] Connected to SSID PCR_sledovani_24 (hostname: irt-esp, IP: 192.168.1.104)
[OTA] Listening to firmware updates on irt-esp.local:8266
[SYSTEM] Disabling serial port communication
Error! Unable to read the iRT bus.
[TELNET] Connected to IRT-ESP version 1.9.10.20110420. Type ? for commands.
log j

System Logging set to Jabber mode

->
(00:00:44.314) irt_rawTelegram: 01: 00
(00:00:44.434) irt_rawTelegram: 01: 01
(00:00:44.554) irt_rawTelegram: 01: 02
(00:00:44.654) irt_rawTelegram: 01: 03
(00:00:44.774) irt_rawTelegram: 01: 00
(00:00:44.895) irt_rawTelegram: 01: 01
(00:00:44.995) irt_rawTelegram: 01: 02
(00:00:45.114) irt_rawTelegram: 01: 03
(00:00:45.235) irt_rawTelegram: 01: 00
(00:00:45.355) irt_rawTelegram: 01: 01
(00:00:45.455) irt_rawTelegram: 01: 02
(00:00:45.575) irt_rawTelegram: 01: 03
(00:00:45.695) irt_rawTelegram: 01: 00
(00:00:45.796) irt_rawTelegram: 01: 01
(00:00:45.915) irt_rawTelegram: 01: 02
(00:00:46.035) irt_rawTelegram: 01: 03
(00:00:46.155) irt_rawTelegram: 01: 00
(00:00:46.256) irt_rawTelegram: 01: 01
(00:00:46.375) irt_rawTelegram: 01: 02
(00:00:46.496) irt_rawTelegram: 01: 03
(00:00:46.596) irt_rawTelegram: 01: 00
(00:00:46.716) irt_rawTelegram: 01: 01
(00:00:46.836) irt_rawTelegram: 01: 02
(00:00:46.956) irt_rawTelegram: 01: 03
(00:00:47.056) irt_rawTelegram: 01: 00
(00:00:47.176) irt_rawTelegram: 01: 01
(00:00:47.296) irt_rawTelegram: 01: 02
(00:00:47.396) irt_rawTelegram: 01: 03
(00:00:47.516) irt_rawTelegram: 01: 00
(00:00:47.636) irt_rawTelegram: 01: 01
(00:00:47.757) irt_rawTelegram: 01: 02
(00:00:47.857) irt_rawTelegram: 01: 03
(00:00:47.977) irt_rawTelegram: 01: 00
(00:00:48.097) irt_rawTelegram: 01: 01
(00:00:48.197) irt_rawTelegram: 01: 02
(00:00:48.318) irt_rawTelegram: 01: 03
(00:00:48.437) irt_rawTelegram: 01: 00
(00:00:48.557) irt_rawTelegram: 01: 01
(00:00:48.657) irt_rawTelegram: 01: 02
(00:00:48.777) irt_rawTelegram: 01: 03
(00:00:48.897) irt_rawTelegram: 01: 00
(00:00:48.997) irt_rawTelegram: 01: 01
(00:00:49.118) irt_rawTelegram: 01: 02
(00:00:49.238) irt_rawTelegram: 01: 03
(00:00:49.358) irt_rawTelegram: 01: 00
(00:00:49.458) irt_rawTelegram: 01: 01
(00:00:49.578) irt_rawTelegram: 01: 02
(00:00:49.698) irt_rawTelegram: 01: 03
(00:00:49.798) irt_rawTelegram: 01: 00
(00:00:49.918) irt_rawTelegram: 01: 01
(00:00:50.038) irt_rawTelegram: 01: 02
(00:00:50.158) irt_rawTelegram: 01: 03
(00:00:50.259) irt_rawTelegram: 01: 00
(00:00:50.379) irt_rawTelegram: 01: 01
(00:00:50.499) irt_rawTelegram: 01: 02
(00:00:50.599) irt_rawTelegram: 01: 03
(00:00:50.719) irt_rawTelegram: 01: 00
(00:00:50.840) irt_rawTelegram: 01: 01
(00:00:50.959) irt_rawTelegram: 01: 02
(00:00:51.059) irt_rawTelegram: 01: 03
(00:00:51.179) irt_rawTelegram: 01: 00
(00:00:51.300) irt_rawTelegram: 01: 01
(00:00:51.400) irt_rawTelegram: 01: 02
(00:00:51.519) irt_rawTelegram: 01: 03
(00:00:51.640) irt_rawTelegram: 01: 00
(00:00:51.760) irt_rawTelegram: 01: 01
(00:00:51.860) irt_rawTelegram: 01: 02
(00:00:51.980) irt_rawTelegram: 01: 03
(00:00:52.100) irt_rawTelegram: 01: 00
(00:00:52.200) irt_rawTelegram: 01: 01
(00:00:52.320) irt_rawTelegram: 01: 02
(00:00:52.440) irt_rawTelegram: 01: 03
(00:00:52.562) irt_rawTelegram: 01: 00
(00:00:52.660) irt_rawTelegram: 01: 01
(00:00:52.781) irt_rawTelegram: 01: 02
(00:00:52.901) irt_rawTelegram: 01: 03
(00:00:53.001) irt_rawTelegram: 01: 00
(00:00:53.122) irt_rawTelegram: 01: 01
(00:00:53.241) irt_rawTelegram: 01: 02
(00:00:53.361) irt_rawTelegram: 01: 03
(00:00:53.461) irt_rawTelegram: 01: 00

How I connected it:
![IMG_20210120_205849](https://user-images.githubusercontent.com/4377405/105228395-d695db80-5b62-11eb-974c-82704a88c5ff.jpg)

**Comment by iamceph** [comment]
This is a log with connected thermostat:
Trying 192.168.1.104...
Connected to 192.168.1.104.
Escape character is '^]'.


* IRT-ESP version 1.9.10.20110420
[FS] System config loaded (509 bytes)
[FS] Custom config loaded (317 bytes)
[WEB] Web server started
[UART] Opened Rx/Tx connection (iRT)
[WIFI] Connecting to SSID: PCR_sledovani_24...
[WIFI] Connected to SSID PCR_sledovani_24 (hostname: irt-esp, IP: 192.168.1.104)
[OTA] Listening to firmware updates on irt-esp.local:8266
[SYSTEM] Disabling serial port communication
Error! Unable to read the iRT bus.
Error! Unable to read the iRT bus.
[TELNET] Connected to IRT-ESP version 1.9.10.20110420. Type ? for commands.
log n

System Logging set to None

->
log j

System Logging set to Jabber mode

->
(00:01:14.125) irt_rawTelegram: 01: 02
(00:01:14.226) irt_rawTelegram: 01: 03
(00:01:14.345) irt_rawTelegram: 01: 00
(00:01:14.806) irt_rawTelegram: 06: 01 80 E0 80 E0 80
(00:01:14.806) irt_otherTelegram: 06: 01 80 E0 80 E0 80
(00:01:15.026) irt_rawTelegram: 03: 01 80 E0
(00:01:15.026) irt_crcErr1: 03: 01 80 E0
(00:01:15.267) irt_rawTelegram: 02: 03 80
(00:01:15.267) irt_crcErr1: 02: 03 80
(00:01:15.827) irt_rawTelegram: 09: 01 80 E0 80 E0 80 00 00 01
(00:01:15.827) irt_otherTelegram: 09: 01 80 E0 80 E0 80 00 00 01
(00:01:16.067) irt_rawTelegram: 03: 02 80 E0
(00:01:16.068) irt_crcErr1: 03: 02 80 E0
(00:01:16.287) irt_rawTelegram: 03: 00 80 C0
(00:01:16.288) irt_crcErr1: 03: 00 80 C0
(00:01:16.868) irt_rawTelegram: 08: 02 80 E0 80 00 01 00 02
(00:01:16.869) irt_otherTelegram: 08: 02 80 E0 80 00 01 00 02
(00:01:16.988) irt_rawTelegram: 01: 03
(00:01:17.209) irt_rawTelegram: 03: 00 80 C0
(00:01:17.209) irt_crcErr1: 03: 00 80 C0
(00:01:17.329) irt_rawTelegram: 01: 02
(00:01:17.549) irt_rawTelegram: 02: 03 80
(00:01:17.549) irt_crcErr1: 02: 03 80
(00:01:18.230) irt_rawTelegram: 0B: 01 80 E0 80 E0 00 00 00 01 00 02
(00:01:18.230) irt_otherTelegram: 0B: 01 80 E0 80 E0 00 00 00 01 00 02
(00:01:18.930) irt_rawTelegram: 0B: 03 80 00 00 01 00 02 00 03 00 00
(00:01:18.931) irt_otherTelegram: 0B: 03 80 00 00 01 00 02 00 03 00 00
(00:01:19.271) irt_rawTelegram: 05: 01 00 02 00 03
(00:01:19.271) irt_otherTelegram: 05: 01 00 02 00 03
(00:01:19.491) irt_rawTelegram: 03: 00 80 01
(00:01:19.492) irt_crcErr1: 03: 00 80 01
(00:01:20.072) irt_rawTelegram: 09: 02 80 E0 00 00 00 01 00 02
(00:01:20.072) irt_otherTelegram: 09: 02 80 E0 00 00 00 01 00 02
(00:01:20.192) irt_rawTelegram: 01: 03
(00:01:20.412) irt_rawTelegram: 03: 00 80 C0
(00:01:20.413) irt_crcErr1: 03: 00 80 C0
(00:01:20.993) irt_rawTelegram: 08: 02 80 E0 80 80 C0 00 02
(00:01:20.994) irt_otherTelegram: 08: 02 80 E0 80 80 C0 00 02
(00:01:21.093) irt_rawTelegram: 01: 03
(00:01:21.333) irt_rawTelegram: 03: 00 80 C0
(00:01:21.334) irt_crcErr1: 03: 00 80 C0

**Comment by jperquin** [comment]
Setting 5 with polling enabled?

**Comment by iamceph** [comment]
> Setting 5 with polling enabled?

Yes

**Comment by Victor-Mo** [comment]
@iamceph  I moved this to issue #13 because it is a different issue.
